src/main/java/com/synopsys/integration/detect/ApplicationUpdater.java

```diff
@@ -307,6 +432,19 @@ private HttpUrl buildDownloadUrl() throws IntegrationException {
         url.append(DOWNLOAD_URL);
         return new HttpUrl(url.toString());
     }
+    
+    private ProxyInfo prepareProxyInfo() {
+        final ProxyInfoBuilder proxyInfoBuilder = new ProxyInfoBuilder();
+        final CredentialsBuilder credentialsBuilder = Credentials.newBuilder();
+        credentialsBuilder.setUsernameAndPassword(proxyProperties.get(ARG_PROXY_USERNAME), 
+                proxyProperties.get(ARG_PROXY_PASSWORD));
+        proxyInfoBuilder.setCredentials(credentialsBuilder.build());
+        proxyInfoBuilder.setHost(proxyProperties.get(ARG_PROXY_HOST));
+        proxyInfoBuilder.setPort(Integer.parseInt(proxyProperties.get(ARG_PROXY_PORT)));
+        proxyInfoBuilder.setNtlmDomain(proxyProperties.get(ARG_PROXY_NTLM_DOMAIN));
+        proxyInfoBuilder.setNtlmWorkstation(proxyProperties.get(ARG_PROXY_NTLM_WORKSTATION));
+        return proxyInfoBuilder.build();
+    }
```

Comment:
DetectConfigurationFactory.createConnectionDetails() does not get called until the boot stage but have we looked at it to see if that code can be reused to establish the proxy details here?

Reply:
The requirements are different. I checked and ruled out code reusability because of how tightly coupled those implementations were within a monolithic Detect. If I go down that path, then a part of Detect would need to be heavily refactored for a unauthenticated API call due to the difference in requirements.

Ideally, this will be addressed when we refactor Detect internal architecture into modular components for configuration, communication and functional core (future looking).

Reply:
Same answer regarding the reusability of DetectPropertyConfiguration.getValue(). The scope of implementation for reuse is much more than the scope of parsing the arguments for specific properties.